src/charm.py

```diff
@@ -590,6 +606,42 @@ def _on_secret_changed(self, event):
     # END: actions
 
     # BEGIN: user management
+    @retry(
+        stop=stop_after_attempt(3),
+        wait=wait_fixed(5),
+        reraise=True,
+        before=before_log(logger, logging.DEBUG),
+    )
+    def _initialise_users(self, event: StartEvent) -> None:
+        """Create users."""
```

Comment:
```suggestion
        """Create users.
        
        User creation can only be completed after the replica set has been initialised which requires some time. In race conditions this can lead to failure to initialise users. To prevent these race conditions from breaking the code, retry on failure.
        """
```